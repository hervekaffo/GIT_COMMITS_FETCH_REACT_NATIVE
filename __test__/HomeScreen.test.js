
import React from 'react';
import { FlatList, Text } from 'react-native';
import renderer from 'react-test-renderer';
import HomeScreen from '../Screens/HomeScreen'

test('renders correctly', () => {
  const tree = renderer.create(<HomeScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the FlatList Component', ()=>{
  const tree = renderer.create(
    <FlatList 
      data={['commit1', 'commit2', 'commit3']}
      keyExtractor={item => item}
      renderItem={({item}) => <Text>{item}</Text>}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});