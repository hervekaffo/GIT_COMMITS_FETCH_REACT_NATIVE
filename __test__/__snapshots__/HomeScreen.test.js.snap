// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Text>
    Home screen
  </Text>
</View>
`;

exports[`renders the FlatList Component 1`] = `
<RCTScrollView
  data={
    Array [
      "commit1",
      "commit2",
      "commit3",
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text>
        commit1
      </Text>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text>
        commit2
      </Text>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text>
        commit3
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
